app/Observers/SymLinkObserver.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace App\Observers;
+
+use App\Facades\Helpers;
+use App\Models\SymLink;
+use Illuminate\Support\Facades\Storage;
+
+class SymLinkObserver
+{
+	/**
+	 * Callback for the SymLink "creating" event.
+	 *
+	 * This method creates the actual symbolic link on disk, before a
+	 * {@link \App\Models\SymLink} is inserted into the database.
+	 * If this method cannot create the symbolic link, then this method
+	 * cancels the insert operation.
+	 *
+	 * @param SymLink $symLink The symbolic link to be created
+	 *
+	 * @throws \Exception Thrown, if this method could not gather enough
```

Comment:
This is very likely to be thrown on the first iterations when you need to generate thousands of symbolic links. :thinking:

Reply:
May be, you are right. But this code hasn't been invented by me. I only moved it around. As I said in the preliminary remark for this PR: while I have been working on this PR I stumbled across a lot of code snippets which probably needs some attention.